frontend/src/main/web/lib/components/glossary/DeleteEntry.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import {PureRenderMixin} from 'react/addons';
+import Actions from '../../actions/GlossaryActions';
+import { Button, Icon, Tooltip, Overlay } from 'zanata-ui';
+
+var DeleteEntry = React.createClass({
+  propTypes: {
+    resId: React.PropTypes.string.isRequired,
+    className: React.PropTypes.string,
+    entry: React.PropTypes.object
+  },
+
+  getInitialState: function() {
+    return {
+      show: false,
+      deleting: false
+    }
+  },
+
+  _handleDelete: function() {
+    this.setState({deleting: true});
+    var self = this;
+    setTimeout(function() {
+      Actions.deleteGlossary(self.props.resId);
+      self._closeDialog();
+    }, 100);
+  },
+
+  _toggleDialog: function () {
+    this.setState({show: !this.state.show});
+  },
+
+  _closeDialog: function () {
+    this.setState(this.getInitialState());
+  },
+
+  render: function () {
+    var self = this, isDeleting = this.state.deleting,
+      info = null;
+
+    if(this.props.entry.termsCount > 0) {
+      info = (<p>
+        Are you sure you want to delete this term and <strong>{this.props.entry.termsCount}</strong> translations?
+        </p>
+      )
+    } else {
+      info =  (<p>
+        Are you sure you want to delete this term?
+      </p>)
+    }
+
+    var tooltip = (<Tooltip id="delete-glossary" title="Delete term and translations">
```

Comment:
Use `const` for anything that is not expected to change. It means we can get help from build tools to make sure we don't change something by mistake, and it means `var` is a special case for something that will be reassigned so we know to be careful with it.